internal/node/launcher.sh

```diff
@@ -259,20 +254,68 @@ if [[ "$PWD" == *"/bazel-out/"* ]]; then
     exit 1
   fi
   EXECROOT=${PWD:0:${index}}
+  RUNFILES_ROOT=$(dirname $PWD)
 else
   # We are in execroot or in some other context all together such as a nodejs_image or a manually
   # run nodejs_binary. If this is execroot then linker node_modules is in the PWD. If this another
   # context then it is safe to assume the node_modules are there and guard that directory if it exists.
   EXECROOT=${PWD}
+  RUNFILES_ROOT=
+fi
+
+# Tell the node_patches_script that programs should not escape the execroot
+export BAZEL_PATCH_ROOT="${EXECROOT}"
+if [[ -n "${VERBOSE_LOGS:-}" ]]; then
+  echo "BAZEL_PATCH_ROOT=${BAZEL_PATCH_ROOT}"
 fi
+
+# Set all bazel managed node_modules directories as guarded so no symlinks may
+# escape and no symlinks may enter.
+# We always guard agains the root node_modules where 1st party deps go.
+# (e.g., /private/.../execroot/build_bazel_rules_nodejs/node_modules)
 export BAZEL_PATCH_GUARDS="${EXECROOT}/node_modules"
-if [[ -n "${BAZEL_NODE_MODULES_ROOT:-}" ]]; then
-  if [[ "${BAZEL_NODE_MODULES_ROOT}" != "${BAZEL_WORKSPACE}/node_modules" ]]; then
-    # If BAZEL_NODE_MODULES_ROOT is set and it is not , add it to the list of bazel patch guards
-    # Also, add the external/${BAZEL_NODE_MODULES_ROOT} which is the correct path under execroot
-    # and under runfiles it is the legacy external runfiles path
-    export BAZEL_PATCH_GUARDS="${BAZEL_PATCH_GUARDS},${BAZEL_PATCH_ROOT}/${BAZEL_NODE_MODULES_ROOT},${PWD}/external/${BAZEL_NODE_MODULES_ROOT}"
+if [[ -n "${BAZEL_NODE_MODULES_ROOTS:-}" ]]; then
+  # BAZEL_NODE_MODULES_ROOTS is in the format "<path>:<workspace>,<path>:<workspace>"
+  # (e.g., "internal/linker/test:npm_internal_linker_test,:npm")
+  if [[ -n "${VERBOSE_LOGS:-}" ]]; then
+    echo "BAZEL_NODE_MODULES_ROOTS=${BAZEL_NODE_MODULES_ROOTS}"
   fi
+  OLDIFS="${IFS}"
+  IFS=","
```

Comment:
What is IFS?

Reply:
bash magic https://bash.cyberciti.biz/guide/$IFS